Dockerfile

```diff
@@ -0,0 +1,23 @@
+# downgrading to python 3.9-slim to avoid issues with faiss-cpu
+FROM python:3.9-slim
+
+# set the working directory in the container
+WORKDIR /app
+
+# copy the dependencies file to the working directory
+COPY requirements.txt .
+
+# install necesary build tools
+RUN apt-get update && \
+    apt-get install -y gcc
```

Comment:
tofix: this will run everytime there are changes in `requirements.txt`

switch this command with `COPY requirements.txt`